### DISCUSSION TITLE
How to pass --tempdir parameter to MACS2 from CallPeaks?

### DISCUSSION THREAD

**Discussion by ajynair** [stuart-lab/signac#1382] [Q&A] [answered]
I am trying Signac pbmc tutorial in Ubuntu and getting errors while using CallPeaks().

>peaks <- CallPeaks(pbmc, macs2.path = "/home/user1/anaconda3/envs/signac/bin/macs2")
..
INFO  @ Tue, 18 Apr 2023 15:07:21: #3 Call peaks... 
INFO  @ Tue, 18 Apr 2023 15:07:21: #3 Pre-compute pvalue-qvalue table... 
INFO  @ Tue, 18 Apr 2023 15:10:11: #3 Call peaks for each chromosome... 
ValueError: cannot resize this array: it does not own its data
Exception ignored in: 'MACS2.IO.CallPeakUnit.clean_up_ndarray'
ValueError: cannot resize this array: it does not own its data
ValueError: cannot resize this array: it does not own its data
Exception ignored in: 'MACS2.IO.CallPeakUnit.clean_up_ndarray'
ValueError: cannot resize this array: it does not own its data
ValueError: cannot resize this array: it does not own its data
Exception ignored in: 'MACS2.IO.CallPeakUnit.clean_up_ndarray'
ValueError: cannot resize this array: it does not own its data
IINFO  @ Tue, 18 Apr 2023 12:52:53: #4 Write output xls file... /tmp/RtmpW3I1Ts/SeuratProject_peaks.xls 
INFO  @ Tue, 18 Apr 2023 12:52:53: #4 Write peak in narrowPeak format file... /tmp/RtmpW3I1Ts/SeuratProject_peaks.narrowPeak 
INFO  @ Tue, 18 Apr 2023 12:52:53: #4 Write summits bed file... /tmp/RtmpW3I1Ts/SeuratProject_summits.bed 
INFO  @ Tue, 18 Apr 2023 12:52:53: Done! 

I am seeing low disk space warning in root disk. First, is this a serious error? Second, I was thinking if MACS2 is working in my /home/user1/tmp (there is space in that disk) this error will go. Can I pass "--tempdir /home/user1/tmp" to the MACS2 from CallPeaks()?

**Comment by timoast** [comment]
By default Signac puts the temporary files in the directory defined by `tempdir()` in R (usually `/tmp/Rxxxxxxxx`). I think the `outdir` parameter in `CallPeaks()` is what you want to change here (there's no `--tempdir` parameter for MACS2 https://github.com/macs3-project/MACS/blob/master/docs/callpeak.md)

Further documentation for `CallPeaks()` is available here: https://stuartlab.org/signac/reference/callpeaks (or run `?CallPeaks` in R)

**Reply by ajynair** [reply, marked as answer]
There is probably some version of MACS2 with a --tempdir (https://github.com/macs3-project/MACS/issues/148) but looks like the one I have installed does not. Anyway, the problem I was facing was MACS2 was writing temporary files to root disk which had space limitations (Signac was writing the temporary files in the tempdir() folder in the user disk so there weren't any issues) and so the number of peaks detected were very low. This was probably due to truncated output from MACS2 due to space limitation. I re-partitioned my root disk to give it more space and the issue is solved. Thanks Tim!
